src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,42 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] checkDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            checkDiagonal[i] = matrix[i][i];
+        }
+        return checkDiagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] checkCounterDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            checkCounterDiagonal[i] = matrix[i][matrix.length - i - 1];
+        }
+        return checkCounterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = matrix[0][0];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                if (matrix[i][j] < min) {
+                    min = matrix[i][j];
+                }
+            }
+        }
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int max = matrix[matrix.length - 1][matrix.length - 1];
```

Comment:
```suggestion
        int max = matrix[0][0];
```
you could've start from the first object... why did you begin from the end?

Reply:
done